lib/TableGen/Operations.cpp

```diff
@@ -127,41 +143,61 @@ void OperationBase::emitArgumentAccessorDefinitions(llvm::raw_ostream &out,
   unsigned numSuperclassArgs = 0;
   if (m_superclass)
     numSuperclassArgs = m_superclass->getNumFullArguments();
-  for (auto indexedArg : llvm::enumerate(m_arguments)) {
+  for (const auto &indexedArg : llvm::enumerate(m_arguments)) {
     const NamedValue &arg = indexedArg.value();
     FmtContextScope scope(fmt);
     fmt.withContext("getContext()");
     fmt.addSubst("index", Twine(numSuperclassArgs + indexedArg.index()));
-    fmt.addSubst("cppType", arg.type->getBuilderCppType());
+
+    if (arg.type->isVarArgList())
+      fmt.addSubst("cppType", arg.type->getVarArgCppGetterType());
+    else
+      fmt.addSubst("cppType", arg.type->getBuilderCppType());
+
     fmt.addSubst("name", arg.name);
     fmt.addSubst("Name", convertToCamelFromSnakeCase(arg.name, true));
 
-    std::string fromLlvm = tgfmt("getArgOperand($index)", &fmt);
-    if (auto *attr = dyn_cast<Attr>(arg.type))
-      fromLlvm = tgfmt(attr->getFromLlvmValue(), &fmt, fromLlvm);
-    else if (arg.type->isTypeArg())
-      fromLlvm += "->getType()";
+    std::string fromLlvm;
+
+    if (!arg.type->isVarArgList()) {
+      fromLlvm = tgfmt("getArgOperand($index)", &fmt);
+      if (auto *attr = dyn_cast<Attr>(arg.type))
+        fromLlvm = tgfmt(attr->getFromLlvmValue(), &fmt, fromLlvm);
+      else if (arg.type->isTypeArg())
+        fromLlvm += "->getType()";
+    } else {
+      fromLlvm = tgfmt("::llvm::make_range((arg_begin() + $index)->get(), "
+                       "(arg_begin() + arg_size() - $index)->get())",
+                       &fmt);
+    }
+
+    fmt.addSubst("fromLlvm", fromLlvm);
+
+    out << tgfmt(R"(
+      $cppType $_op::get$Name() {
+        return $fromLlvm;
+      })",
+                 &fmt);
+
+    // We do not generate a setter for variadic arguments for now.
+    if (arg.type->isVarArgList())
+      continue;
```

Comment:
Maybe split this into `generateGetter()` and `generateSetter()`? Then, this loop reduces to

```
generateGetter();
if (!arg.type->isVarArgList())
  generateSetter();
```

which may be a bit clearer.

Reply:
Will clean up, thanks.